### PR TITLE
fixed [Docs] Spot/interruptible docs imply retries come from the user… Closes #3956

### LINKED ISSUE #3956: [Docs] Spot/interruptible docs imply retries come from the user retry budget
### Description

In the [spot/interruptible docs](https://docs.flyte.org/projects/cookbook/en/latest/auto_examples/containerization/spot_instances.html#setting-interruptible) we have the following note
> If you set retries=n, for instance, and the task gets preempted repeatedly, Flyte will retry on a preemptible/spot instance n-1 times and for the last attempt will retry your task on a non-spot (regular) instance. Please note that tasks will only be retried if at least one retry is allowed using the retries parameter in the task decorator.

This implies that interruptible retries come from the user-specified user error retry budget. However, in practice spot interruptions are taken from the system retry budget.

### Are you sure this issue hasn't been raised already?

- [X] Yes

### Have you read the Code of Conduct?

- [X] Yes

### PR DESCRIPTION
**Pull request by ap0calypse8**
… retry budget #3956

## What changes were proposed in this pull request?

This PR clarifies the documentation regarding retry behavior for interruptible tasks in Flyte. It updates sections to emphasize that retries for spot/interruptible instances utilize the system retry budget rather than the user-specified retry budget. Changes are made in the user guide and related documentation files to ensure accurate guidance for users managing task retries on preemptible compute resources.

### CODE REVIEW

File: docs/user_guide/concepts/main_concepts/tasks.rst
Comment (davidmirror-ops) [inline docs/user_guide/concepts/main_concepts/tasks.rst:152]: Great catch. 
```suggestion
Starting from 1.10.0, Flyte offers a simplified retry behavior where both system and user retries count towards a single retry budget defined in the task decorator. To enable this:
```

**Review by davidmirror-ops — APPROVED** [review]

File: docs/user_guide/concepts/main_concepts/tasks.rst
Comment (ap0calypse8) [reply docs/user_guide/concepts/main_concepts/tasks.rst]: Hey I updated what you said . Please review my changes.

**Review by davidmirror-ops — DISMISSED** [review]

**Comment by ap0calypse8** [comment]
@davidmirror-ops  hey can you help me out on what is wrong? so that i can fix the issue asap. thanks!

**Comment by ap0calypse8** [comment]
I am waiting for workflow approval from maintainers @davidmirror-ops . Thanks!

**Comment by davidmirror-ops** [comment]
Next time you can also install requirements and build locally with make -c <path to your docs folder> html to catch syntax errors

**Comment by ap0calypse8** [comment]
@davidmirror-ops  As i have updated the syntax of the docs ....unit test flytectl failing has no relation to my commit.this test failure might be:

A flaky test
An existing issue in the master branch
An environmental issue in CI 

help me out please

**Comment by ap0calypse8** [comment]
@davidmirror-ops  PLease help me out as I have done the test but for flytetcl i cound not find any issues

**Review by davidmirror-ops — APPROVED** [review]
Thank you!

**Comment by ap0calypse8** [comment]
@davidmirror-ops  thank you

**Pull request merged**
